CMakeLists.txt

```diff
@@ -713,10 +713,8 @@ set(GMLC_NETWORKING_CONCURRENCY_INCLUDE ${PROJECT_SOURCE_DIR}/ThirdParty/concurr
 set(GMLC_NETWORKING_DISABLE_ASIO ${HELICS_DISABLE_ASIO})
 
 if(HELICS_ENABLE_ENCRYPTION)
-    set(HELICS_ENABLE_ENCRYPTED_COMMS 1)
```

Comment:
This step was happening after the helics-config.h file is already generated without the define -- also seemed a bit redundant having an extra variable with the same meaning.

Reply:
The config.h does use this, so we might need to move this up in the CMAKE so it gets used by the config.h generation.  Since I think the code does need to know if the encrypted option is even available.

Reply:
Have you verified that option turns on and off properly?

Reply:
I changed it because before it was always showing the error message about HELICS not being compiled without encryption support, even when the option was turned on.

In the CI builds there are a mix of Linux jobs that have the option ON/OFF, and earlier builds for this PR printed the libraries linked to one of the HELICS binaries.

gcc11 build linked with OpenSSL: https://dev.azure.com/HELICS-test/HELICS/_build/results?buildId=11666&view=logs&jobId=0cbb91d6-5794-5751-6296-8904318eba52&j=0cbb91d6-5794-5751-6296-8904318eba52&t=f7b3e03b-d64b-5c39-d720-8c777666cf2a

clang13 build without OpenSSL: https://dev.azure.com/HELICS-test/HELICS/_build/results?buildId=11666&view=logs&jobId=0cbb91d6-5794-5751-6296-8904318eba52&j=ec8c292a-0c4e-5923-6a96-214ddcb87113&t=0bd14658-bf30-5e59-496c-3881c52cd135